kong/runloop/balancer/init.lua

```diff
@@ -346,6 +346,15 @@ local function set_host_header(balancer_data)
     upstream_host = upstream_host .. ":" .. balancer_data.port
   end
 
+  -- the nginx grpc module does not offer a way to override the
+  -- :authority pseudo-header; use our internal API to do so
+  if upstream_scheme == "grpc" or upstream_scheme == "grpcs" then
```

Comment:
Should we only call `kong.service.request.set_header(":authority", upstream_host)` when `upstream_host ~= orig_upstream_host`? It seems to be unnecessary to call it for every retry if `upstream_host` didn't change.

Reply:
Good catch. If the host did not change, there is no reason to call it. Thanks!